lib/oli/delivery/attempts/auto_submit/worker.ex

```diff
@@ -2,35 +2,119 @@ defmodule Oli.Delivery.Attempts.AutoSubmit.Worker do
+          }
+
+          Logger.info("Auto submit: got resource attempt for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+          case Graded.finalize(context) do
+            {:ok, %FinalizationSummary{resource_access: resource_access, part_attempt_guids: part_attempt_guids, lifecycle_state: :evaluated}} ->
+
+              Logger.info("Auto submit: finalized as :evaluated for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+              section = Sections.get_section_by(slug: section_slug)
+              user = Oli.Accounts.get_user!(resource_access.user_id)
+
+              Oli.Delivery.Snapshots.Worker.perform_now(part_attempt_guids, section_slug)
+
+              Logger.info("Auto submit: snapshots created for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+              if section.grade_passback_enabled do
+                Logger.info("Auto submit: gradepassback STARTED for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+                result = Oli.Grading.send_score_to_lms(
+                  section,
+                  user,
+                  resource_access,
+                  Oli.Delivery.Attempts.PageLifecycle.GradeUpdateWorker.access_token_provider(section)
+                )
+                Logger.info("Auto submit: gradepassback FINISHED for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+                result
+              else
+                Logger.info("Auto submit: gradepassback NOT ENABLED for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+                :ok
+              end
+
+            {:ok, _} ->
+              Logger.info("Auto submit: finalized not as :evaluated for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+              :ok
+
+            {:error, :already_submitted} ->
+
+              Logger.info("Auto submit: was already finalized for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+              section = Sections.get_section_by(slug: section_slug)
+              user = Oli.Accounts.get_user!(resource_access.user_id)
+
+              if section.grade_passback_enabled do
+
+                Logger.info("Auto submit: gradepassback STARTED for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+                result = Oli.Grading.send_score_to_lms(
+                  section,
+                  user,
+                  resource_access,
+                  Oli.Delivery.Attempts.PageLifecycle.GradeUpdateWorker.access_token_provider(section)
+                )
+
+                Logger.info("Auto submit: gradepassback FINISHED for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+                result
+              else
+                Logger.info("Auto submit: gradepassback NOT ENABLED for guid #{resource_attempt_guid} and section slug #{section_slug}")
+
+                :ok
+              end
```

Comment:
Should we still send the score to the lms in this case?

Reply:
Yes, this was intentional.  Imagine the situation where auto submit finalizes the attempt BUT then the grade passback fails.  This will fail the entire job, which will be retried.  In the second attempt we will get the `{:error, :already_submitted}` when we call `finalize`, which we handle, and then take another crack at grade sync.